tools/tests/modal/test.js

```diff
@@ -28,3 +41,28 @@ describe('Modal Component', () => {
         assert(modal.style.display === 'none', 'The modal is not hidden.');
     });
 });
+
+/* global engine */
+/* global modal */
+if (engine?.isAttached) {
+    describe('Modal Component (Gameface Data Binding Test)', () => {
+        const template = `<div data-bind-for="array:{{modal.array}}"><gameface-modal></gameface-modal></div>`;
+
+        afterAll(() => cleanTestPage('.test-wrapper'));
+
+        beforeEach(function (done) {
+            engine.createJSModel('modal', { array: [{}, {}] });
+            engine.updateWholeModel(modal);
+
+            setupModal(template).then(done).catch(err => console.error(err));
+
+            engine.synchronizeModels();
+        });

```

Comment:
```suggestion
        beforeEach(function (done) {
            engine.createJSModel('modal', { array: [{}, {}] });
            engine.updateWholeModel(modal);

            setupModal(template).then(done).catch(err => console.error(err));

            engine.synchronizeModels();
        });

```

This code is repeated throughout the tests. Maybe make it a util that accepts template, name of the model and model and just call it somewhere.

Ex:
```
async setupTest (template, modelName, model = { array: [{}, {}] }) {
         engine.createJSModel(modelName, model);
            engine.updateWholeModel(window[modelName]);

             await setup(template)
            engine.synchronizeModels();
}
```

Reply:
Added `setupDataBindingTest` in **actions.js** and updated tests.
**Note:** I'm passing the function that creates + inserts the element / component on the page.